packages/core/admin/admin/src/StrapiApp.js

```diff
@@ -33,36 +33,67 @@ const queryClient = new QueryClient({
   },
 });
 
-// FIXME
 const appLocales = Object.keys(translations);
 
 class StrapiApp {
   constructor({ appPlugins }) {
     this.appPlugins = appPlugins || {};
-    this.componentApi = Components();
-    this.fieldApi = Fields();
-    // FIXME
-    this.middlewares = [];
+    this.library = Library();
+    this.middlewares = Middlewares();
     this.plugins = {};
-    this.reducers = { ...reducers };
+    this.reducers = Reducers({ appReducers: reducers });
     this.translations = translations;
-    // this.addMiddleware = this.addMiddleware.bind(this);
-    this.getPlugin = this.getPlugin.bind(this);
   }
 
-  addMiddleware(middleware) {
-    this.middlewares.push(middleware);
-  }
+  addComponent = component => {
+    this.library.components.add(component);
+  };
+
+  addComponents = components => {
+    components.map(compo => this.library.components.add(compo));
+  };
```

Comment:
WDYT about having only one function that could handle one or multiple components?

```js
addComponents = components => {
   if (Array.isArray(components) {
       components.map(c => add(c))
   } else {
       add(components)
   }
}
```

Same for fields, middlewares, reducers